### PR TITLE
`select-enrollment-channel` now accepts protocol-defined inputs

### PR DESCRIPTION
**Pull request by jaredperreault-okta**

### CODE REVIEW

File: test/spec/idx/authenticate.ts [hunk 1]
```diff
@@ -2027,6 +2027,71 @@ describe('idx/authenticate', () => {
...
+            label: 'Phone Number',
+            required: true,
+            type: 'string',
+          });
+
+          await proceed(authClient, {
+            phoneNumber: '+1234'
+          });
+
+          expect(enrollmentChannelDataSmsResponse.proceed).toHaveBeenCalled();
```
Comment (shuowu-okta) [inline test/spec/idx/authenticate.ts:2093]: use toHaveBeenCalledWith to verify payload?

File: test/spec/idx/authenticate.ts [hunk 2]
```diff
@@ -2027,6 +2027,71 @@ describe('idx/authenticate', () => {
...
+            step: 'select-enrollment-channel'
+          });
+          const { options } = res.nextStep!;
+          expect(options).toContainEqual({
+            label: 'SMS',
+            value: 'sms'
+          });
+
+          res = await proceed(authClient, {
+            // channel: 'phoneNumber'
```
Comment (shuowu-okta) [inline test/spec/idx/authenticate.ts:2069]: remove?

File: lib/idx/remediators/SelectEnrollmentChannel.ts [hunk 3]
```diff
@@ -25,7 +26,18 @@ export class SelectEnrollmentChannel extends Remediator<SelectEnrollmentChannelV
...
 
   canRemediate() {
-    return Boolean(this.values.channel);
+    if (this.values.channel) {
+      return true;
+    }
+
+    if (this.values.authenticator) {
+      const { id, channel } = this.values.authenticator as Authenticator;
+      if (!!id && !!channel) {
```
Comment (shuowu-okta) [inline lib/idx/remediators/SelectEnrollmentChannel.ts:35]: `||` instead of `&&`?

Comment (jaredperreault-okta) [reply lib/idx/remediators/SelectEnrollmentChannel.ts:35]: both the `channel` and `id` are required

Comment (shuowu-okta) [reply lib/idx/remediators/SelectEnrollmentChannel.ts:35]: nvm, I miss read `!!` to `!` ...

**Review by shuowu-okta — APPROVED** [review]

**Pull request merged**